provider/internal/connections/common.go

```diff
@@ -99,3 +108,148 @@ func attrValue(ctx context.Context, val attr.Value) (interface{}, error) {
 
 	return nil, fmt.Errorf("unsupported type %T", val)
 }
+
+func clearSensitiveValuesFromRead(attrs map[string]schema.Attribute, config map[string]any) map[string]any {
+	for k, v := range attrs {
+		switch v := v.(type) {
+		case schema.ListNestedAttribute:
+			if v.IsSensitive() {
```

Comment:
If wonder if this would be a little clearer/easier to understand if the `v.IsSensitive` part wasn't repeated in each branch; something like

```go
for k, v := range attrs {
  if v.IsSensitive() { 
    delete(config, k)
    continue
  }
  switch v := v.(type) { 
    ...
  }
}
```